From 0a0cf394929b3aec2747c201d8d9d74172392bc8 Mon Sep 17 00:00:00 2001
From: Melissa Wen <mwen@igalia.com>
Date: Tue, 29 Jul 2025 20:44:34 -0100
Subject: [PATCH 026/161] drm/amd/display: use dcn10 helper to translate
 in_shaper_lut on dcn32

it's the same used by dcn20 and dcn30 for plane shaper LUT and have some
differences. cm_helper doesn't support HLG TF, but we are not using it
too.

Signed-off-by: Melissa Wen <mwen@igalia.com>
---
 drivers/gpu/drm/amd/display/dc/hwss/dcn32/dcn32_hwseq.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/hwss/dcn32/dcn32_hwseq.c b/drivers/gpu/drm/amd/display/dc/hwss/dcn32/dcn32_hwseq.c
index 51fd2b3fe..959a153e1 100644
--- a/drivers/gpu/drm/amd/display/dc/hwss/dcn32/dcn32_hwseq.c
+++ b/drivers/gpu/drm/amd/display/dc/hwss/dcn32/dcn32_hwseq.c
@@ -506,10 +506,9 @@ bool dcn32_set_mcm_luts(
 		lut_params = &plane_state->cm.shaper_func.pwl;
 	else if (plane_state->cm.shaper_func.type == TF_TYPE_DISTRIBUTED_POINTS) {
 		// TODO: dpp_base replace
-		rval = cm3_helper_translate_curve_to_hw_format(plane_state->ctx,
-			&plane_state->cm.shaper_func,
-			&dpp_base->shaper_params,
-			true);
+		rval = cm_helper_translate_curve_to_hw_format(plane_state->ctx,
+						       &plane_state->cm.shaper_func,
+						       &dpp_base->shaper_params, true);
 		lut_params = rval ? &dpp_base->shaper_params : NULL;
 	}
 
-- 
2.56.0

